test/integration/certificates/generates_new_private_key_per_request_test.go

```diff
@@ -78,7 +78,7 @@ func TestGeneratesNewPrivateKeyIfMarkedInvalidRequest(t *testing.T) {
 	}
 
 	var firstReq *cmapi.CertificateRequest
-	if err := wait.Poll(time.Millisecond*500, time.Second*10, func() (done bool, err error) {
+	if err := wait.PollUntilContextTimeout(context.TODO(), time.Millisecond*500, time.Second*10, true, func(ctx context.Context) (bool, error) {
```

Comment:
suggestion: It looks like we actually have a context here and so we wouldn't need to use `context.TODO()`. Can we use the `ctx` that already exists?